lib/ansible/plugins/inventory/aws_ec2.py

```diff
@@ -307,6 +307,11 @@ def _get_instances_by_region(self, regions, filters, strict_permissions):
 
         for connection, region in self._boto3_conn(regions):
             try:
+                # This is important if hostnames are used other than IP addresses; once a host
+                # is terminated those attributes no longer exist, but others may:
+                # if an instance-state-name filter has not been specified, find the running instances
+                if not any([f['Name'] == 'instance-state-name' for f in filters]):
+                    filters.append({'Name': 'instance-state-name', 'Values': ['running']})
```

Comment:
I'm not sure we want to assume running. I can see never collecting terminated/shutting-down instances because instances in those states will never be available https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-lifecycle.html 

I think a better default would be to collect pending/running/stopping/stopped hosts, since all of those may become available during a play, or have host vars that are needed.